hadoop-yarn-project/hadoop-yarn/hadoop-yarn-server/hadoop-yarn-server-common/src/test/java/org/apache/hadoop/yarn/server/federation/utils/TestFederationStateStoreFacade.java

```diff
@@ -232,4 +239,111 @@ public void testGetApplicationHomeSubClusterCache() throws YarnException {
     }
   }
 
+  @Test
+  public void testStoreNewMasterKey() throws YarnException, IOException {
+    // store delegation key;
+    DelegationKey key = new DelegationKey(1234, 4321, "keyBytes".getBytes());
+    HashSet<DelegationKey> keySet = new HashSet<DelegationKey>();
+    keySet.add(key);
+    facade.storeNewMasterKey(key);
+
+    MemoryFederationStateStore federationStateStore =
```

Comment:
Does it need to be MemoryFederationStateStore?

Reply:
This part of the code is written because only MemoryFederationStateStore is currently implemented, but it is not very clean to write this way, I will refactor this part of the code.